stdlib/tests/crypto/stark/verifier_recursive/mod.rs

```diff
@@ -24,6 +24,62 @@ pub struct VerifierData {
+/// We also include the expected layout of the operand stack for completness.
+///
+/// # Operand Stack Layout
+///
+/// ```
+/// | Position | Content           | Size (in Felt) | Description                      |
+/// | -------- | ----------------- | -------------- | -------------------------------- |
+/// | 0        | grinding_factor   | 1              | Proof-of-work difficulty in bits |
+/// | 1        | num_queries       | 1              | Number of FRI queries            |
+/// | 2        | trace_length_log2 | 1              | Log2 of execution trace length   |
+/// ```
+///
+/// # Advice Stack Layout
+///
+/// ```
+/// | Position | Content                | Size (in Felt) | Description                                           |
+/// | -------- | ---------------------- | -------------- | ----------------------------------------------------- |
+/// | 0        | variable_len_pi_size   | 1              | Size of variable length PI in Felt                    |
+/// | 1-n      | public_inputs_data     | varies         | Input/output stacks + Program digest + kernel digests |
+/// | n+1-n+4  | aux_randomness         | 4              | β = (β₀, β₁), ɑ = (ɑ₀, ɑ₁) (filled later)             |
+/// | n+5      | num_kernel_procedures  | 1              | Count of kernel procedure digests                     |
+/// | n+6-m    | trace_commitments      | 4 or 8         | Main/auxiliary trace segment commitments              |
+/// | m+1-p    | constraint_commitment  | 4              | Constraint composition commitment                     |
+/// | p+1-p+2  | alpha_deep_placeholder | 2              | Deep composition randomness (filled later)            |
+/// | p+3-q    | ood_evaluations        | varies         | Out-of-domain trace and constraint evals              |
+/// | q+1-r    | fri_commitments        | varies         | FRI layer commitment digests                          |
+/// | r+1-s    | fri_remainder_poly     | varies         | FRI remainder polynomial coefficients                 |
+/// | s+1      | pow_nonce              | 1              | Proof-of-work nonce                                   |
+/// ```
+///
+/// # Advice Merkle Store Content
+///
+/// ```
+/// |         Merkle trees         |
+/// | ---------------------------- |
+/// | Main trace segment tree      |
+/// | Auxiliary trace segment tree |
+/// | Constraint composition tree  |
+/// | FRI layers commitment trees  |
+/// ```
+///
+/// # Advice Map Layout
+///
+/// ```
+/// | Key (Word)                     | Value (Vec<Felt>)                   |
+/// | ------------------------------ | ----------------------------------- |
+/// | Query index                    | Main trace evaluations              |
+/// | Query index                    | Auxiliary trace evaluations         |
+/// | Query index                    | Constraint evaluations              |
+/// | FRI layer (folded) query index | FRI codeword evaluations on a coset |
```

Comment:
Are we actually using query indexes s keys? I thought we'd be using leaf hashes as keys and leaf pre-images as values.

If we are using indexes, how can the same index be used for multiple values (i.e., for main trace, aux trace, and constraint evaluations).

Reply:
Absolutely correct, this should be the leaf at the query index. Thanks for catching it.
Fixed it now